### ISSUE TITLE
No need for prevState

### ISSUE THREAD

**Issue by KhalilKhunji** [Shahstaa/controlled-forms-in-react-lab#2]
https://github.com/Shahstaa/controlled-forms-in-react-lab/blob/2b55d43d4fcaf1f6312db8b339e7bcb84711e469/src/Bookshelf.jsx#L15-L19
https://github.com/Shahstaa/controlled-forms-in-react-lab/blob/2b55d43d4fcaf1f6312db8b339e7bcb84711e469/src/Bookshelf.jsx#L24

The [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) (...) can already create a copy of the `book` state, so that you aren't mutating it directly.
So something like:
```
setNewBook({...newBook, [name] : value});
```
where `{...newBook` is copying the newBook object,
or
```
setBooks([...books, newBook]);
```
Would already be enough, since the `[...books` portion makes a copy of the `books` array. `prevState` here is redundant.

Basically, the spread operator allows you to make a copy of arrays, objects, strings, etc. Which is useful since it means we can make a copy of a state (e.g. `book` or `newBook`) and then perform operations on it without modifying the original state, and after that set a new state.
